File: 0350-Intersection of Two Arrays II/0350-Intersection of Two Arrays II.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        unordered_map<int, int> table1;
       
        for (int num : nums1) {
            ++table1[num];
        }
        
        vector<int> result;
        for (int num : nums2) {
            if(table1[num]-- > 0) {
                result.push_back(num);
            }
        }

        return result;
    }
};

File: 0301-0400/0350-Intersection of Two Arrays II/0350-Intersection of Two Arrays II.py
import collections
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        table = collections.Counter(nums1)
        result = []
        for num in nums2:
            if table[num] > 0:
                table[num] -= 1
                result.append(num)
        return result
